### PR TITLE
board/arm/nrf52xx_boards: NFCT pins configuration

### PR DESCRIPTION
**Pull request by nvlsianpu**
It is possible to use NFCT pins as regular GPIOs.
This patch introduce option for enabling this feature.

Fix for #6937 

Additionally this fix nrf52 part of #7452

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

File: boards/arm/nrf52840_pca10056/Kconfig [hunk 1]
```diff
@@ -10,4 +10,15 @@ config GPIO_AS_PINRESET
 	bool "GPIO as pin reset (reset button)"
 	default y
 
+config NFCT_PINS_AS_GPIOS
+	bool "NFCT pins acts as GPIOs"
+	default n
```
Comment (anangl) [inline boards/arm/nrf52840_pca10056/Kconfig]: There is no need to specify it, it is set implicitly.

File: boards/arm/nrf52840_pca10056/Kconfig [hunk 2]
```diff
@@ -10,4 +10,15 @@ config GPIO_AS_PINRESET
 	bool "GPIO as pin reset (reset button)"
 	default y
 
+config NFCT_PINS_AS_GPIOS
+	bool "NFCT pins acts as GPIOs"
```
Comment (anangl) [inline boards/arm/nrf52840_pca10056/Kconfig]: ... pins act ...
Or maybe omit "act"?

**Review by anangl — COMMENTED** [review]
What about pca10040?

**Review by carlescufi — APPROVED** [review]
Looks good.

**Comment by anangl** [comment]
Shouldn't the same be done for pca10040 as well? I thought this PR would be a good place for this.

**Comment by nvlsianpu** [comment]
Initially I was thinking about adding NFCT_PINS_AS_GPIOS keyword to the arch/arm/soc/nordic_nrf/nrf52/Kconfig.soc file, but options form this file are available in menuconfig via [ARM Options]->[SoC Options]->*. That why I add another Kconfig.board file.

File: arch/arm/soc/nordic_nrf/nrf52/Kconfig.soc [hunk 3]
```diff
@@ -31,7 +31,7 @@ endchoice
 
 config ARM_MPU_NRF52X
 	bool "Enable MPU on nRF52"
-	depends on CPU_HAS_MPU
+	depends on CPU_HAS_MPU && SOC_SERIES_NRF52X
```
Comment (galak) [inline arch/arm/soc/nordic_nrf/nrf52/Kconfig.soc]: Drop this change from this PR, should be in a separate PR.

**Review by galak — CHANGES_REQUESTED** [review]
Drop the MPU change from this PR.

**Comment by nvlsianpu** [comment]
@galak done

**Review by anangl — APPROVED** [review]

File: arch/arm/soc/nordic_nrf/nrf52/Kconfig.board [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Kconfig.board - Nordic Semiconductor nRF52 MCU line
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+config GPIO_AS_PINRESET
```
Comment (galak) [inline arch/arm/soc/nordic_nrf/nrf52/Kconfig.board]: We should put these in  arch/arm/soc/nordic_nrf/nrf52/Kconfig.soc as the code that implements them exists in arch/arm/soc/nordic_nrf/nrf52/soc.c

**Review by galak — CHANGES_REQUESTED** [review]
Rather than introducing a new arch/arm/soc/nordic_nrf/nrf52/Kconfig.board and sourcing it from boards.  Lets put these Kconfig symbols in the Kconfig.soc as that is where the code that implements them exists.

**Comment by nvlsianpu** [comment]
@galak I moved both GPIO_as_pin_reset and NFCT_as_GPIO options to the Kconfig.soc

Image below shows how it is looks like now:
image

**Review by galak — APPROVED** [review]
